test/tests/userobjects/openmc_cell_transform/rotate_cells.i

```diff
@@ -0,0 +1,145 @@
+[Mesh]
+  [pin]
+    type = ConcentricCircleMeshGenerator
+    num_sectors = 6
+    radii = '1.0'
+    rings = '3 3'
+    has_outer_square = on
+    pitch = 4.0
+    preserve_volumes = yes
+    smoothing_max_it = 3
+  []
+  [assembly]
+    type = CartesianIDPatternedMeshGenerator
+    inputs = 'pin'
+    pattern = '0 0;
+               0 0'
+    assign_type = 'cell'
+    id_name = 'pin_id'
+  []
+  [delete]
+    type = BlockDeletionGenerator
+    input = assembly
+    block = 2
+  []
+  [extrude]
+    type = MeshExtruderGenerator
+    input = delete
+    extrusion_vector = '0 0 30'
+  []
+  [translate]
+    type = TransformGenerator
+    input = extrude
+    transform = TRANSLATE_CENTER_ORIGIN
+    vector_value = '-12 12 0'
+  []
+[]
+
+
+[GlobalParams]
+  displacements = 'disp_x disp_y disp_z'
+  use_displaced_mesh = true
+[]
+
+[Problem]
+  type = OpenMCCellAverageProblem
+  cell_level = 2
+  verbose = true
+  power = 100
+  fixed_mesh = false
+
+  [Tallies]
+    [heat]
+      type = CellTally
+      score = 'kappa_fission'
+      block = '1'
+    []
+  []
+[]
+
+[AuxVariables]
+  [disp_x]
+    [AuxKernel]
+        type = FunctionAux
+        variable = disp_x
+        function = disp_x_fn
+        execute_on = 'timestep_begin'
+        use_displaced_mesh = false
+    []
+  []
+  [disp_y]
+    [AuxKernel]
+        type = FunctionAux
+        variable = disp_y
+        function = disp_y_fn
+        execute_on = 'timestep_begin'
+        use_displaced_mesh = false
+    []
+  []
+  [disp_z]
+  []
+[]
+
+[Functions]
+  [psi_fn]
+    type = ParsedFunction
+    expression = '5*t'
+  []
+  [disp_x_fn]
+    type = ParsedFunction
+    expression = (cos(5*t*3.14159265359/180)-1)*(x-x0)-sin(5*t*3.14159265359/180)*(y-y0)
+    symbol_names = 'x0 y0'
+    symbol_values = '0 0'
+  []
+  [disp_y_fn]
+    type = ParsedFunction
+    expression = sin(5*t*3.14159265359/180)*(x-x0)+(cos(5*t*3.14159265359/180)-1)*(y-y0)
+    symbol_names = 'x0 y0'
+    symbol_values = '0 0'
+  []
```

Comment:
Do you plan to change `x0` and `y0` ever? Maybe not really necessary to include them and just put the 0s in the `expression` line?